### PR TITLE
merge all configs in pyproject.toml

### PR DESCRIPTION
**Pull request by Borda**
as `pyproject.toml` already exists in this repo and it is the new python standard, it makes sense to pull all particular configurations and drop redundant `setup.cfg`.

As Flake8 does not support `pyproject.toml` I have moved all essential arguments to the single call in CI :rabbit:

### CODE REVIEW

**Comment by Borda** [comment]
@ogrisel @lesteve @tomMoral may I ask you for feedback? :rabbit:

**Comment by Borda** [comment]
friendly ping @ogrisel @lesteve @tomMoral :flamingo:

**Comment by tomMoral** [comment]
This is a nice PR. We will keep `setup.cfg` for now to make it easy to run `flake8` on the project but this is definitely a step in the right direction. Maybe it would make sense to move to `ruff` to drop `flake8` and thus `setup.cfg`. PR welcome if you have the time :)

We just duplicated the PR in #1460 to be able to merge this before the release today.

**Comment by Borda** [comment]
> PR welcome if you have the time :)

I feel confused when you welcome PR, but instead of adjusting mine/existing, you create a new one with another author... that is not a good way how to motivate anyone to contribute more, sorry :disappointed:

**Comment by Borda** [comment]
@fcharras @tomMoral so switch the flake8 t ruff in this PR or in another? doing it here would be trivial, just though to keep a PR minimal and focus

File: setup.cfg [hunk 1]
```diff
@@ -1,18 +1,4 @@
...
-doc_files = doc
-
-[tool:pytest]
-doctest_optionflags = NORMALIZE_WHITESPACE ELLIPSIS
-addopts =
-    --doctest-glob="doc/*.rst"
-    --doctest-modules
-    --ignore joblib/externals
-testpaths = joblib
+# The remaining setting as Flake8 does not support pyproject.tom, so for any config use  pyproject.toml
```
Comment (tomMoral) [inline setup.cfg:1]:
```suggestion
# The prefered config file is pyproject.toml. The use of setup.cfg is
# mostly for compatibility with flake8 so it should not be used if possible.
```

**Review by tomMoral — APPROVED** [review]
LGTM

**Comment by Borda** [comment]
@tomMoral @fcharras thank you and apology if I was to straight... :rabbit:

**Comment by Borda** [comment]
the failing docs build seems unrelated and appears on multiple PRs

**Comment by tomMoral** [comment]
Yes this was fixed in #1461 , the PR should now be green for merge

**Pull request merged**